system-variables: Add charset documentation (#5867)
pingcap/docs: system-variables.md

- Default value: `ON`
- Controls whether statements should automatically commit when not in an explicit transaction. See [Transaction Overview](/transaction-overview.md#autocommit) for more information.

### character_set_client

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`

> - Character set for data send by the client. See [Character Set and Collation](/character-set-and-collation.md) for details on the use of character sets and collations in TiDB. It is recommended to use [`SET NAMES`](/sql-statements/sql-statement-set-names.md) to change the character set when needed.

Reviewer: ```suggestion
- The character set for data sent from the client. See [Character Set and Collation](/character-set-and-collation.md) for details on the use of character sets and collations in TiDB. It is recommended to use [`SET NAMES`](/sql-statements/sql-statement-set-names.md) to change the character set when needed.
```

### character_set_connection

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`

> - Character set for string literals that don't have a character set specified.

Reviewer: ```suggestion
- The character set for string literals that do not have a specified character set.
```

### character_set_database

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`

> - This indicates the character set of the default database that is in use. Setting this variable is not recommended. The server changes this varible when a new default database is selected.

Reviewer: ```suggestion
- This variable indicates the character set of the default database in use. **It is NOT recommended to set this variable**. When a new default database is selected, the server changes the variable value.
```

### character_set_results

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`
- The character set that is used when data is sent to the client.

### character_set_server

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`

> - The collation used for new schemas in case none is provided by the `CREATE SCHEMA` statement.

Reviewer 2: ```suggestion
- The character set used for new schemas in case none is provided by the `CREATE SCHEMA` statement.
```

> - The character set used for new schemas in case none is provided by the `CREATE SCHEMA` statement.

Reviewer: ```suggestion
- The character set used for new schemas when no character set is specified in the `CREATE SCHEMA` statement.
```

- Default value: `ON`
- Controls whether statements should automatically commit when not in an explicit transaction. See [Transaction Overview](/transaction-overview.md#autocommit) for more information.

### character_set_client

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`
- Character set for data send by the client. See [Character Set and Collation](/character-set-and-collation.md) for details on the use of character sets and collations in TiDB.

### character_set_connection

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`
- Character set for string literals that don't have a character set specified.

### character_set_database

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`
- This indicates the character set of the default database that is in use. Setting this variable is not recommended. The server changes this varible when a new default database is selected.

### character_set_filesystem

- Scope: SESSION | GLOBAL
- Default value: `binary`

> - This variable is not used and exists for compatibility with MySQL.

Reviewer 3: We don't currently document any of the noop sysvars. I suggest we omit these, as I hope that in the next sprint we will also offer a feature to disable them (if it doesn't go ahead, we can maybe revisit them after all the actual sysvars are documented.)

Author: Does `CharacterSetFilesystem` fall into that category?

It is defined in `sysvar.go` and not in `noop-.go` like `CharacterSetSystem`.

https://github.com/pingcap/tidb/blob/ac5cf972159822928b3e9ee1b848e7f4fe0a8d0d/sessionctx/variable/sysvar.go#L682-L683

https://github.com/pingcap/tidb/blob/ac5cf972159822928b3e9ee1b848e7f4fe0a8d0d/sessionctx/variable/noop.go#L157

Reviewer 3: Yes. It looks like it should actually be in `noop.go`.

Author: Ok. I'll remove variables defined in `noop.go` from this.
I'll also remove `character_set_filesystem`.
